library/src/main/java/com/ironz/binaryprefs/encryption/XorKeyEncryptionImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.ironz.binaryprefs.encryption;
+
+import com.ironz.binaryprefs.exception.EncryptionException;
+
+import java.util.Arrays;
+
+/**
+ * @author Sergey Boishtyan
+ */
+
+public final class XorKeyEncryptionImpl implements KeyEncryption {
+
+    private static final int KEY_LENGTH = 16;
+
+    private final byte[] xor;
+
+    public XorKeyEncryptionImpl(byte[] xor) {
+        this.xor = xor;
+        checkLength();
+        checkMirror();
+    }
+
+    @Override
+    public String decrypt(String name) {
+        return xorName(name);
+    }
+
+    @Override
+    public String encrypt(String name) {
+        return xorName(name);
+    }
+
+    private String xorName(String name) {
+        byte[] beforeXor = name.getBytes();
+        int beforeXorLength = beforeXor.length;
+        int xorLength = xor.length;
+        byte[] afterXor = new byte[beforeXorLength];
+        for (int beforeXorIndex = 0; beforeXorIndex < beforeXorLength; beforeXorIndex++) {
+            byte afterXorItem = beforeXor[beforeXorIndex];
+            for (int xorIndex = 0; xorIndex < xorLength; xorIndex++) {
```

Comment:
I think inner loop should be as an separate method. Also this method should take raw byte and return xor'ed byte. In other words inner loop and inner loop indexes code parts should be fully incapsulated.